docs/TypedMongo.md

```diff
@@ -0,0 +1,446 @@
+```
+  
+One of the consequences of the above is that `Task` is 
+[referentially transparent](https://en.wikipedia.org/wiki/Referential_transparency) while `Future` is not.
+For people acknowledged with (pure) functional programming this will be a virtue by itself. Here we can outline some
+immediate practical benefits of that property.
+
+One of the consequences of `Task`'s referential transparency that makes it much easier to work with than with `Future` 
+is that we don't need an implicit `ExecutionContext` to invoke transformation methods like `map`, `flatMap`, etc. 
+Instead, a `Scheduler` (extended `ExecutionContext`) is necessary but only at the very point where `Task` is being
+executed and converted into a `Future` (among other options).
+
+This means that we can compose our asynchronous programs (represented as `Task`s) out of smaller programs without
+being bound to any particular `Scheduler`. This relieves us from constant dragging of an `ExecutionContext` via
+implicit parameters that we have to do when working with `Future`s.
+
+```scala
+val fetchValue: Task[Int] = ...
+def convertValue(int: Int): Task[String] = ...
+
+// no ExecutionContext/Scheduler necessary at this point
+val fullProgram: Task[Unit] = 
+  for {
+    value <- fetchValue
+    converted <- convertValue(value)
+  } yield {
+    println(converted)
+  }
+
+// only now we need a Scheduler
+import monix.execution.Scheduler.Implicits.global
+val future: Future[Unit] = fullProgram.runToFuture
+```
+
+### [Monix `Observable`](https://monix.io/docs/current/reactive/observable.html)
+
+An `Observable[T]` is conceptually the same thing as a Reactive Streams `Publisher[T]` (actually, it's fairly straightforward
+to convert between each other). That is, both represent an asynchronous stream of values of type `T` with backpressure
+(a consumer must explicitly request more elements from the producer). The difference is that `Observable` exposes a 
+rich, Scala-idiomatic API with a lot of high-level combinators.
+
+There are many ways to consume results of an `Observable[T]`. If you don't need the streaming nature of the `Observable`,
+you can simply "degrade" it onto a `Task[List[T]]` by calling `.toListL` on it. This is very often used in MongoDB API
+in order to fetch full results of a query into memory.
+
+**NOTE**: You can think of `Observable` as an asynchronous version of `Iterable`.
+
+### Blocking
+
+It is recommended to use MongoDB API in a non-blocking way (by composing `Task`s and `Observables`s).
+However, sometimes this is not possible - some external API may force us into synchronous, blocking implementations.
+Then we have no choice but to wait for the database on the current JVM thread.
+
+In order to do this, use utilities provided by `com.avsystem.commons.concurrent.BlockingUtils` class. Your project should
+provide an implementation of this class (often a globally accessible object) where an appropriate Monix `Scheduler`s
```

Comment:
Can we **not** recommend using an `object` for this? Statically defined thread pools usually mean 0 runtime configuration for this.

Reply:
Well, I have actually used a globally accessible object for this...
I'm currently under impression that thread pools are one of the rare things that can be kept in globals because in order to  make sure that they are reused throughout your _entire process_. The whole point of using asynchronous API is to reduce the number of threads to minimum.

If we need compatibility then... JVM properties? (I would not think I'd ever say that).

Reply:
I know it's tempting, but injecting some object which provides blocking capabilities **and** is a Scheduler / EC / TP is not that much of a hassle and makes (us) developers reconsider the way we run async tasks. Globally accessible object can also be messy in tests, it's very hard to control or mock it then.

Reply:
I've removed the "globally accessible object" suggestion.